### ISSUE TITLE
go vet can silently fail in go 1.10.1 

### ISSUE THREAD

**Issue by sronderos-atlassian** [alecthomas/gometalinter#486]
I encountered a discrepancy between gometalinter output in go 1.9.1 and go 1.10.2.

go vet was reporting an error when running gometalinter with 1.9.1, but did not report with go 1.10.2

In the debug output for 1.10.2 I noticed the following
```
DEBUG: [Jun  8 14:22:25.299] [vet.1]: executing /usr/local/bin/go vet . ./app ./lib/handlers ./systest
DEBUG: [Jun  8 14:22:25.618] [vet.1]: warning: /usr/local/bin/go returned exit status 1: can't load package: package <...>/systest: build constraints exclude all Go files in <...>/systest

DEBUG: [Jun  8 14:22:25.618] [vet.1]: vet hits 0: <...>
```
It seems that `go vet` will return a warning if it encounters a package in which all the files were excluded by build flags. In this project there is a single go file in /systest and it contains the build tag `// +build integration`. There are quite a few workarounds that I could find: 

1. Add an empty go file to /systest
1. Override vet in the metalinter config with `"Command": "go vet -tags integration"`
1. Override vet in the metalinter config with `"PartitionStrategy": "single-directory"`
1. Run gometalinter with go 1.9.1

I think the most appropriate thing would be to have the linter fail if `go vet` returns a warning if `go` fails due to build constraints.

**Comment by alecthomas** [comment]
Apparently go vet now relies on the build cache to operate correctly (or at all?). As gometalinter doesn't install anything. This is fixed in golangci-lint (https://github.com/golangci/golangci-lint/issues/87).
